### ISSUE TITLE
Show the loading screen each time with a different picture

### ISSUE THREAD

**Issue by ShadesOT** [We-the-People-civ4col-mod/Mod#174]
This issue is a follow up to #158 . For information and context read #158 first.

**Comment by Nightinggale** [comment]
I modified the MainMenu xml file to contain a list of Loading strings. Whenever CvMainMenuInfo::getLoading() is called, it returns a random one from the list.

Luckily the exe asks for the file each time it has to show the loading screen, meaning you do not have to restart the game to get a new random image.

Technically I cheated regarding the random. I had issues getting the asynchronous random to work and get a proper random seed. Instead I ended up not using random at all, instead I use the number of milliseconds since system start. This means rather than random, it acts like it loops the array sequentially and you just happen to get whatever is displayed when you click something, which will need a loading screen. At a rate of 1000 images per second, it sure feels random which one it will display. Funny how using the clock feels more random than using random with a not completely random random seed.

I'm not closing this ticket because while the DLL supports multiple images, the xml files still only contain one. In other words we need more paintings.

**Comment by ShadesOT** [comment]
> In other words we need more paintings.

Will do.

Meanwhile: The size of .dds files is breathtaking. We should decide how many MB we want to add, to have more variety in background images.
1920x1440 (which gives full resolution at 1920x1080): 8 MB/image (ultra zipped 6.7 MB)
1200x900 (which looks still nice on 1920x1080): 3.2 MB/image (ultra zipped 2.8 MB)
1024x768 (which still looks ok-ish at 1920x1080): 2.3 MB/image (ultra zipped 2.0 MB)

I am heading for the nice image quality, which cost max 2.8 MB/image compressed. 

Maybe I con lower the download size by consolidating the existing images. It appears to me there are ones which are not used in the game.

**Comment by Nightinggale** [comment]
It's a bit hard to judge quality without the images. Perhaps you could add an image here at various resolutions, which would then allow the rest of us to get an opinion on quality vs size.

`
Maybe I can lower the download size by consolidating the existing images. It appears to me there are ones which are not used in the game.
`
A bit off topic, but it would be a good idea to make some sort of graphic verification system where the xml files are scanned for which images the game uses and then alert on files, which appears to be unused. For all we know we might have a lot of unused files.

**Comment by ShadesOT** [comment]
> would be a good idea to make some sort of graphic verification system where the xml files are scanned for which images the game uses

That verification system would need to scan .nif files as well (see https://github.com/We-the-People-civ4col-mod/Mod/commit/aafe80be0b9126aebb58b1c634cf5bc6f795dae2).

**Comment by ShadesOT** [comment]
Here is the comparrison ... 

1920 wide
![landscape1920x1080](https://user-images.githubusercontent.com/43323245/46918469-ebbe5000-cfd2-11e8-98c0-3fb2ed45ac10.jpg)

1200 wide
![landscape1200x675](https://user-images.githubusercontent.com/43323245/46918480-0264a700-cfd3-11e8-9849-b344744d207f.jpg)

1024 wide
![landscape1024x576](https://user-images.githubusercontent.com/43323245/46918485-085a8800-cfd3-11e8-9d20-a1afcab2f042.jpg)

**Comment by Nightinggale** [comment]
I can clearly see an improvement for each step. While I from a quality point of view prefer the 1920 one, I'm not sure I like it 5 MB more. I don't think it's on the screen often enough to justify bloating the size of the download.

**Comment by LibSpit** [comment]
yeah even before seeing it I was thinking the 3.2Mb one was 'good enough'. The only other think to suggest would be an optional 'HD Package' for people who need the 1080+ world to stay sharp at all times.

**Comment by Nightinggale** [comment]
If we are a bit more clever, we can make the game autogenerate the list of possible backgrounds at startup instead of defining them in xml. This way we can make a standalone package people can add, which is not just high resolution, but also adds more screens.

I don't feel like coding it right away, but from a technical point of view, it shouldn't be hard. The CivEffect branch even has a function in CvGlobals, which is called post xml loading for setting up various caches. It can be used for something like this as well as it will generate more xml entries/data after loading the xml files, but before using them.
